docs/sources/operators-guide/deploy-grafana-mimir/migrate-to-unified-proxy-deployment/_index.md

```diff
@@ -0,0 +1,344 @@
+
+The unification also brings new features to the GEM gateway: OpenShift Route and horizontal autoscaling of the
+gateway Pods.
+
+The unified configuration lives in the `gateway` section of the values file. With this we also
+deprecate the `nginx` section. It will be removed in `mimir-distributed` version 7.0.0.
+
+It is possible to migrate from `nginx` to the `gateway` configuration without downtime too. The migration should take
+less than 30 minutes. The rest of this article contains a procedure for migrating from the old `nignx` section to
+`gateway`.
+
+## Before you begin
+
+Make sure that the version of the `mimir-distributed` Helm chart that you have installed is 4.0.0 or higher.
+
+## Procedure
+
+1. Scale out the `gateway` deployment
+
+   1. Change your Helm values file to enable the `gateway` and increase its replicas. Set the number of replicas of
+      the gateway Deployment to the number of
+      replicas that NGINX are running with. For example, if you have deployed 10 NGINX replicas, then
+      add the following to your Helm values file `custom.yaml`:
+
+      ```yaml
+      gateway:
+        enabledNonEnterprise: true
+        replicas: 10
+      ```
+
+   2. Deploy your changes.
+
+      ```bash
+      helm upgrade $RELEASE grafana/mimir-distributed -f custom.yaml
+      ```
+
+2. Replace the `nginx` deployment with `gateway`.
+
+   1. Disable NGINX. Add or merge the following with your values file:
+
+      ```yaml
+      nginx:
+        enabled: false
+      ```
+
+   2. If you are using the Ingress that the chart provides, then copy the `ingress` section from `nginx` to
+      `gateway` and override the name. Override the name to the name of the Ingress resource that
+      the chart created for NGINX or the GEM gateway.
+
+      Reusing the name allows the `helm` command to retain the existing resource instead of deleting it and
+      recreating it under a slightly different name.
+
+      In the example that follows, the name of the Ingress
+      resource was `mimir-nginx`. Use `kubectl` to get the name of the existing Ingress resource:
+
+      ```bash
+      kubectl get ingress
+      ```
+
+      ```console
+      NAME          CLASS    HOSTS               ADDRESS    PORTS     AGE
+      mimir-nginx   <none>   mimir.example.com   10.0.0.1   80, 443   172d
+      ```
+
+      After carrying out this step the Helm values for `gateway` should look like the following:
+
+      ```yaml
+      gateway:
+        ingress:
+          enabled: true
+          nameOverride: mimir-nginx
+          hosts:
+            - host: mimir.example.com
+              paths:
+                - path: /
+                  pathType: Prefix
+          tls:
+            - secretName: mimir-gateway-tls
+              hosts:
+                - mimir.example.com
+        enabledNonEnterprise: true
+        replicas: 10
+      ```
+
+   3. Copy the `service` section from `nginx` to `gateway` and override the name. Override the name to
```

Comment:
I think I get it. I need to add nameOverride if I'm using the service directly. In my case I use the service as my grafana datasource, but I use the ingress as the url in my browser, so I needed to update both to avoid downtime.

Reply:
that's right. I also added a sentence to make it clear why we are doing this